src/parsers/manifestjson.js

```diff
@@ -797,6 +801,13 @@ export default class ManifestJSONParser extends JSONParser {
     }
   }
 
+  validateHomePageURL(url) {
+    const restrictedDomain = 'addons-dev.allizom.org';
+    if (url.indexOf(restrictedDomain) !== -1) {
+      this.collector.addError(messages.RESTRICTED_HOMEPAGE_URL);
+    }
```

Comment:
I think that we should also mark the manifest as invalid (by setting this.isValid to false as we do in a few other places in `src/parsers/manifestjson.js`).